linux_os/guide/system/accounts/accounts-session/accounts_user_dot_no_world_writable_programs/oval/shared.xml

```diff
@@ -1,52 +1,73 @@
 
-  <local_variable id="var_accounts_user_dot_no_world_writable_programs_dirs"
-                  datatype="string" version="1"
-                  comment="Variable including all home dirs from interactive users">
+  <local_variable id="var_accounts_user_dot_no_world_writable_programs_dirs" datatype="string"
+      version="1" comment="Variable including all home dirs from interactive users">
     <object_component item_field="home_dir"
-                      object_ref="object_accounts_user_dot_no_world_writable_programs_objects"/>
+      object_ref="object_accounts_user_dot_no_world_writable_programs_objects"/>
   </local_variable>
 
-  <!-- #### creation of object #### -->
-  <unix:file_object id="object_accounts_user_dot_no_world_writable_programs_init_files"
-                    version="1">
+  <local_variable id="var_world_writable_programs" datatype="string" version="1"
+      comment="Variable including all world writable programs">
+    <object_component item_field="filepath" object_ref="object_world_writable_programs"/>
+  </local_variable>
+
+  <local_variable id="var_world_writable_programs_regex" datatype="string" version="1"
+      comment="Regex including all world writable programs">
+    <concat>
+      <literal_component>^[^#]*</literal_component>
+      <escape_regex>
+        <variable_component var_ref="var_world_writable_programs"/>
+      </escape_regex>
+    </concat>
+  </local_variable>
+
+  <unix:file_state id="state_world_writable_programs" version="1" operator='AND'>
+    <unix:type>regular</unix:type>
+    <unix:owrite datatype="boolean">true</unix:owrite>
+  </unix:file_state>
+
+  <unix:file_object id="object_world_writable_programs" version="1">
     <unix:behaviors recurse="directories" recurse_direction="down" max_depth="-1"
-                    recurse_file_system="local"/>
-    <unix:path var_ref="var_accounts_user_dot_no_world_writable_programs_dirs"
-               var_check="at least one"/>
-    <unix:filename operation="pattern match">^\..*</unix:filename>
+      recurse_file_system="local"/>
+    <unix:path>/</unix:path>
+    <unix:filename operation="pattern match">^.*$</unix:filename>
+    <filter action="include">state_world_writable_programs</filter>
   </unix:file_object>
 
-  <!-- #### creation of state #### -->
-  <unix:file_state id="state_accounts_user_dot_no_world_writable_programs" version="1"
-                   operator='AND'>
-    <unix:gwrite datatype="boolean">false</unix:gwrite>
-    <unix:owrite datatype="boolean">false</unix:owrite>
-  </unix:file_state>
+  <ind:textfilecontent54_object id="object_accounts_user_dot_no_world_writable_programs_init_files"
+      version="1">
+    <ind:behaviors recurse="directories" recurse_direction="down" max_depth="-1"
+      recurse_file_system="local"/>
+    <ind:path var_ref="var_accounts_user_dot_no_world_writable_programs_dirs"
+      var_check="at least one"/>
+    <ind:filename operation="pattern match">^\..*</ind:filename>
```

Comment:
There is a problem with using the textfilecontent54_object matching any file whose name starts with a dot (matching `^\..*`). Some of the files are binary, for example `.esd_auth` which is often present on systems with GUI. These files can't be applied a regular expression on because that works only on text files. The pattern match will fail and the evaluation of this rule will end with an error. I would prefer to have a specific list of files here instead, eg. `.bashrc`, `.zshrc`.

Reply:
Introduced a variable to define the initialization files to search for.
I deemed this approach better than having some hardcoded value on the OVAL file.